src/Shared/PlatformNegotiation.cs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.BackEnd;
+
+namespace Microsoft.Build.Shared
+{
+    /// <summary>
+    /// This class contains only static methods, which are useful throughout many
+    /// of the XMake classes and don't really belong in any specific class.   
+    /// </summary>
+    static internal class PlatformNegotiation
+    {
+        internal static string? GetPlatformString(String projectReferencePlatformMetadata, String projectReferenceLookupTableMetadata, String CurrentProjectPlatformMetadata, String CurrentPlatformLookupTableMetadata, String ProjectPath, TaskLoggingHelper? Log = null)
+        {
+            
+            Dictionary<string, string>? currentProjectLookupTable = ExtractLookupTable(CurrentPlatformLookupTableMetadata, Log);
+
+            if (string.IsNullOrEmpty(projectReferencePlatformMetadata))
+            {
+                if(Log != null)
+                {
+                    Log.LogWarningWithCodeFromResources("GetCompatiblePlatform.NoPlatformsListed", ProjectPath);
+                }
```

Comment:
For these log statements, you can write:

```suggestion
                Log?.LogWarningWithCodeFromResources("GetCompatiblePlatform.NoPlatformsListed", ProjectPath);
```

Reply:
Much cleaner, thank you